lib/rules/define-macros-order.js

```diff
@@ -130,6 +138,17 @@ function create(context) {
       },
       onDefineExposeExit(node) {
         defineExposeNode = getDefineMacrosStatement(node)
+      },
+
+      'CallExpression:exit'(node) {
+        if (
+          node.callee &&
+          node.callee.type === 'Identifier' &&
+          order.includes(node.callee.name) &&
+          !KNOWN_MACROS.has(node.callee.name)
+        ) {
+          macrosNodes.set(node.callee.name, [getDefineMacrosStatement(node)])
```

Comment:
I think we need to check whether the CallExpression is located in a place that is accepted as a macro.
For example, I don't think the following definePage is a macro.

```vue
<script setup>
function fn() {
  definePage()
}
</script>
```

Reply:
I'm wondering how to do that 🤔 I guess macros are only located at the "root" of the script?

Reply:
Even if macros could be used in places other than the root, I don't think this rule would allow we to change their order.

Reply:
I made a simple rule where the node needs to be an `ExpressionStatement` or a `VariableDeclaration`.
```js
 const macro = getDefineMacrosStatement(node)
 if (
  macro.type === 'ExpressionStatement' ||
  macro.type === 'VariableDeclaration'
 )
   macrosNodes.set(node.callee.name, [macro])
```
But I don't know if it's a good approach. I added a test so that 
```js
<script setup>
function fn() {
    definePage()
}
</script>
```
is considered valid (`definePage` is ignored), but I have an issue with this test : 
```vue
<script setup>
import Foo from 'foo'
/** model */
const model = defineModel()
/** emits */
defineEmits(['update:foo'])

const val = () => definePage()
</script>
```
This one keeps failing because I don't know how to ignore the "macro".

---
There is a `getDefineMacrosStatement` that is simply a function that searches recursively for the `Program` node; maybe it could be more "precise"? 
Because macros (I believe):
- Have the `Program` as parent
- Are expressions that are called
- Can be assigned like `const props = defineProps<...>()`
- Are located directly on the first column, or just after an assignment

Maybe it is more precise, but these are some ideas to target more precisely what "macros" are within the AST 🙂

Reply:
Hmm. I've never used custom macros, so I'm not familiar with them.
If you want to prioritize support for just `definePage`, I think it's reasonable for now to only support `Program > ExpressionStatement > CallExpression (custom macro)`.

Reply:
Instead of `CallExpression`, I added:
- `Program > ExpressionStatement > CallExpression` to target things like `defineProps()`, `definePage()`, `defineSomething()`...
- `Program > VariableDeclaration > VariableDeclarator > CallExpression` to target `const props = defineProps()`, `const emit = defineEmits()`, `const stuff = defineStuff()`...